addons/missileguidance/script_component.hpp

```diff
@@ -16,3 +16,14 @@
 #endif
 
 #include "\z\ace\addons\main\script_macros.hpp"
+
+#define RANDOM_VECTOR_3D (call {\
+    private _z = random 2 - 1;\
+    private _r = sqrt (1 - _z^2);\
+    private _theta = random 360;\
+    [_r * cos _theta, _r * sin _theta, _z]\
+})
```

Comment:
Archimedes!

Reply:
What do you mean?

Reply:
This random distribution function is based on the fact that a sphere and a cylinder with height = radius have the same surface area. Something that Archimedes figured out.

Reply:
Well don't give me the credit for coming up with it, give it to yourself :P https://github.com/acemod/ACE3/pull/6708#discussion_r237926302

Reply:
Hmm, maybe I should create it as a CBA function. It seems to be used in multiple places now.

Reply:
@commy2 Create it as a function, but generalise it to `n` dimensions ;)